src/lib/state/state.svelte.ts

```diff
@@ -64,11 +67,67 @@ export const globalState = $state<GlobalState>({
     // Indicates if user is using existing approval process, creating one or injected provider
     approvalType: 'existing',
 
+    constructorArguments: {
+      values: {},
+      required: 0
+    },
+
+    deterministic: {
+      isSelected: false,
+      salt: undefined
+    },
+
     // Indicates if deployment is completed.
     completed: false,
   },
 });
 
+export const isValidFormAuthentication = () => globalState.authenticated
+
+const isFormFilledFor = (formInputToCheck: keyof typeof globalState.form) => () => globalState.form[formInputToCheck] !== undefined
+export const isValidFormNetwork = isFormFilledFor("network")
+
+export const isValidConstructorArguments = () => Object.keys(globalState.form.constructorArguments.values).length === globalState.form.constructorArguments.required && Object.values(globalState.form.constructorArguments.values).reduce((checkValueResult: boolean, currentValue) => checkValueResult && (currentValue !== undefined && currentValue !== ""), true)
+
+export const isValidDeterministicConfiguration = () => globalState.form.deterministic.isSelected ? globalState.form.deterministic.salt !== undefined : true
+
+export const isValidFormApprovalProcess = async () => {
```

Comment:
There is an issue with the following scenario:
1. Select a Network that is not yet part of a Deploy Environment
2. Select "Create New Approval Process"
3. Select EOA
4. Enter Address
5. Hover over Deploy tab -- notice the mouse icon shows it is disabled
6. Click anywhere -- UI proceeds to the Deploy tab
7. Go back to the Approval Process tab -- "Use existing Approval Process" is selected instead of "Create New Approval Process"

See recording:
![approvalprocess](https://github.com/user-attachments/assets/60cf7bb0-957d-47ee-ba17-9ed633aaef5b)